src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java

```diff
@@ -176,7 +176,11 @@ private void cancelClicked()
      */
     private void confirmClicked()
     {
+        if (building.getBuildingLevel() > 0 && !building.getStyle().equals(styles.get(stylesDropDownList.getSelectedIndex())))
```

Comment:
Maybe we should here:
a) check if the hut is deconstructed
b) Send a message telling the player it failed?

Reply:
forgot b) I saw the code down there now